main.py

```diff
@@ -121,21 +141,26 @@ def main() -> None:
     # Read input
     user_inputs = _parse_env_vars()
 
-    # Execute action
-    discourse = create_discourse(
-        hostname=user_inputs.discourse.hostname,
-        category_id=user_inputs.discourse.category_id,
-        api_username=user_inputs.discourse.api_username,
-        api_key=user_inputs.discourse.api_key,
-    )
-    urls_with_actions_dict = run(
-        base_path=pathlib.Path(),
-        discourse=discourse,
-        user_inputs=user_inputs,
-    )
+    # Run this only if the reference to discourse exists
+    clients = get_clients(user_inputs, pathlib.Path())
+
+    # Open a PR with community contributions if necessary
+    migrate_result = execute_in_tmpdir(
+        lambda tmp_path: run_migrate(
```

Comment:
I don't have strong opinion, but just to double check I understand this correctly (together with the following comment): you mean something like

```python
@execute_in_tmpdir
def main_migrate(repo_path, user_inputs):
    clients = get_clients(user_inputs, repo_path)
    if clients.repository.metadata.docs:
        return run_migrate(clients, user_inputs
    else:
        return {}
```

If so, I need to refactor a bit more `execute_in_tmpdir`, as it needs to change its signature to also deal with functions with other inputs (should this be generic to handle multiple inputs?), e.g. 

```
def execute_in_tmpdir(Callable[[Path, A], B]) -> Callable[A, B]
```

Also, if we want to place the code above in `__init__.py`, we need to move also `execute_in_tmpdir`, otherwise we will end up with circular dependency (but I'm not really sure it is best practice to put so much code in `__init__.py` either),

Reply:
I think this is fine, I was thinking the 

```python
    if clients.repository.metadata.docs:
        return run_migrate(clients, user_inputs
    else:
        return {}
```

Could go to `run_migrate`, i.e., just get it to return `{}` `if not clients.repository.metadata.docs`. Does that make sense?